### PR TITLE
Convert simple_select_list_paginate_txn to return tuples.

### PR DESCRIPTION
**Pull request by clokep**
More of my war on `cursor_to_dict`. Follow-u to #16431. This targets a single "simple" method: `simple_select_list_paginate_txn` and its two callers.

### CODE REVIEW

File: synapse/federation/send_queue.py [hunk 1]
```diff
@@ -395,7 +395,7 @@ class PresenceDestinationsRow(BaseFederationRow):
     @staticmethod
     def from_data(data: JsonDict) -> "PresenceDestinationsRow":
         return PresenceDestinationsRow(
-            state=UserPresenceState.from_dict(data["state"]), destinations=data["dests"]
+            state=UserPresenceState(**data["state"]), destinations=data["dests"]
```
Comment (clokep) [inline synapse/federation/send_queue.py:398]: I didn't really need to nuke this, but it seemed like a useless method?

File: synapse/api/presence.py [hunk 2]
```diff
@@ -81,8 +81,18 @@ def as_dict(self) -> JsonDict:
...
+    ) -> "UserPresenceState":
+        return UserPresenceState(
+            user_id=row[0],
+            state=row[1],
+            last_active_ts=row[2],
+            last_federation_update_ts=row[3],
+            last_user_sync_ts=row[4],
+            status_msg=row[5],
+            currently_active=bool(row[6]),
+        )
```
Comment (erikjohnston) [inline synapse/api/presence.py:95]: I'm concerned that this is so far away from where we declare the ordering of the columns in the SQL. Can we inline this?

Comment (clokep) [reply synapse/api/presence.py:95]: Inline it meaning -- return it directly from the datastore?

Comment (clokep) [reply synapse/api/presence.py:95]: There's only one use of this anyway, wonder why I bothered w/ a helper...

**Review by erikjohnston — APPROVED** [review]

**Pull request merged**